### ISSUE TITLE
[Question] Namespace conflict error with mulitbase with a common base example

### ISSUE THREAD

**Issue by deepak-gc** [kubernetes-sigs/kustomize#4056]
Hi, 
I followed the multibase with common base [example ](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/multibases).

```.
├── base
│   ├── kustomization.yaml
│   └── pod.yaml
├── dev
│   └── kustomization.yaml
├── kustomization.yaml
├── namespace.yaml
├── production
│   └── kustomization.yaml
```

Everything works as expected when i run the command 

```> kustomize build```

I also want to be able to deploy only dev || production 

```> kustomize build dev``` 

```> kustomize build production``` 

If I add a namespace in the dev & production overlays then the above commands works but then the command to build dev and production deployment fails with error 

`may not add resource with an already registered Namespace`

Is there anyway this is achievable using Kustomize?

**Comment by natasha41575** [comment]
> If I add a namespace in the dev & production overlays then the above commands works but then the command to build dev and production deployment fails with error

Sorry I didn't quite follow. In dev and production overlays, are you adding another namespace.yaml file or are you adding a `namespace` field to the kustomization? Could you provide the contents of the files you are adding/changing?

What commands work or fail? I'm unsure what you mean by the command do build dev and production deployment.

**Comment by deepakgc** [comment]
@natasha41575 Thank you for your reply!, please see the code and the Readme file in this repo https://github.com/deepak-gc/kustomize-namespace-issue. If it's still not clear enough please let me know.

**Comment by deepak-gc** [comment]
Hi @natasha41575  On single namespace on prod env where I would like to deploy same app multiple times with different deployment names.
my kustomize dir:
```
├── k8s
│   ├── base
│   │   ├── deployment.yaml
│   │   └── kustomization.yaml
│   └── overlays
│       └── prod
│           ├── kustomization.yaml
│           ├── namespace-a
│           │   ├── deployment-a1
│           │   │   ├── kustomization.yaml
│           │   │   ├── namespace.yaml
│           │   │   └── patch.yaml
│           │   ├── deployment-a2
│           │   │   ├── kustomization.yaml
│           │   │   └── patch.yaml
│           │   ├── kustomization.yaml
│           │   └── namespace.yaml
```

In `k8s/overlays/prod/namespace-a/kustomization.yaml` 
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

namespace: namespace-a

resources:
  - namespace.yaml
  - deployment-a1
  - deployment-a2
```

So I can create deployment for  `deployment-a1` & `deployment-a2` using below command

`kustomize build k8s/overlays/prod/namespace-a`

which works flawlessly :) 

But I also would like to be able to create a deployment  only for `deployment-a` 

`kustomize build k8s/overlays/prod/namespace-a/deployment-a`

So I added a namespace.yaml in `k8s/overlays/prod/namespace-a/deployment-a1/` and included it in `k8s/overlays/prod/namespace-a/deployment-a1/kustomization.yaml`  

after this I am able to create deployment for only 'deployment-a' but the previous command 

`kustomize build k8s/overlays/prod/namespace-a`

no loger works because we have 2 namnespace.yaml with same name and error is thrown.

As per my requirement which is a single namespace, multiple deployment and be able to create deployment for all or only one as I have specified above. I'll appreciate any suggestion for the directory structure or may be creating namespace first with separate kustomize build...  or anything to improve this.

I also tried with components(added namespace.yaml) there but ended up with same error after I added the component in each deployment kuustomization.yaml
